exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/managed/SortConfig.java

```diff
@@ -114,15 +127,25 @@ public SortConfig(DrillConfig config) {
     } else {
       mSortBatchSize = Character.MAX_VALUE;
     }
+
+    // Disable the single-batch shortcut; primarily for testing
+
+    if (config.hasPath(ExecConstants.EXTERNAL_SORT_OPTIMIZE_SINGLE_BATCH)) {
+      enableSingleBatchShortcut = config.getBoolean(ExecConstants.EXTERNAL_SORT_OPTIMIZE_SINGLE_BATCH);
+    } else {
```

Comment:
Yes. But the "final" provides useful information: the value can't change once set. Once we accept "final" then we're stuck with the awkward code to initialize the value only once.